store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -51,6 +54,7 @@
   private Expression filterExpression;
   private String tableName;
   private Configuration hadoopConf;
+  private boolean useVectorReader;
```

Comment:
I suggest we can make vector as default as we see benefit in the performance.  
withVectorReader() can be a deprecated to support just for disable  in case of issues till the time vector reader is stabilized. @ravipesala , is the fine?

Reply:
ok, And also make sure that complex datatypes should be switched to normal record reader automatically